csharp/src/Ice/Proxy.cs

```diff
@@ -2115,39 +2066,13 @@ public static ObjectPrx checkedCast(ObjectPrx b)
     /// <param name="b">The proxy to cast to ObjectPrx.</param>
     /// <param name="ctx">The Context map for the invocation.</param>
     /// <returns>b.</returns>
-    public static ObjectPrx checkedCast(ObjectPrx b, Dictionary<string, string> ctx)
-    {
-        return b;
```

Comment:
This optimization was not correct (see #199). When you can checkedCast, you want to see an ice_isA call.